go/vt/vtgate/planbuilder/testdata/select_cases.json

```diff
@@ -3542,7 +3542,7 @@
         "Table": "`user`, dual"
       },
       "TablesUsed": [
-        "main.dual",
+        "user.dual",
```

Comment:
This change seems OK to me, potentially an improvement. If you don't agree I can make code changes to preserve the existing plan.